### PR TITLE
Update some threading docs with xml comments

### PR DESCRIPTION
**Pull request by kouvel**
Updated APIs in the `System.Threading` namespace for https://github.com/dotnet/runtime/issues/54866

### CODE REVIEW

File: xml/System.Threading/CancellationToken.xml [hunk 1]
```diff
@@ -778,11 +778,23 @@ For the definition of equality, see the <xref:System.Threading.CancellationToken
...
+        <param name="state">The state to pass to the <paramref name="callback" /> when the delegate is invoked.  This may be <see langword="null" />.</param>
+        <summary>Registers a delegate that will be called when this <see cref="T:System.Threading.CancellationToken">CancellationToken</see> is canceled.</summary>
+        <returns>The <see cref="T:System.Threading.CancellationTokenRegistration" /> instance that can be used to unregister the callback.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+If this token is already in the canceled state, the delegate will be run immediately and synchronously. Any exception the delegate
+            generates will be propagated out of this method call. The current <xref:System.Threading.ExecutionContext">ExecutionContext</see>, if one exists,
```
Comment (gewarren) [inline xml/System.Threading/CancellationToken.xml]:
```suggestion
generates is propagated out of this method call. The current <xref:System.Threading.ExecutionContext>, if one exists,
```

File: xml/System.Threading/CancellationToken.xml [hunk 2]
```diff
@@ -778,11 +778,23 @@ For the definition of equality, see the <xref:System.Threading.CancellationToken
...
+            will be captured along with the delegate and will be used when executing it. The current <xref:System.Threading.SynchronizationContext> is not captured.
```
Comment (gewarren) [inline xml/System.Threading/CancellationToken.xml:792]:
```suggestion
is captured along with the delegate and is used when executing it. The current <xref:System.Threading.SynchronizationContext> is not captured.
```

File: xml/System.Threading/CancellationToken.xml [hunk 3]
```diff
@@ -1015,11 +1027,22 @@ End If
...
+        <param name="state">The state to pass to the <paramref name="callback" /> when the delegate is invoked.  This may be <see langword="null" />.</param>
+        <summary>Registers a delegate that will be called when this <see cref="T:System.Threading.CancellationToken">CancellationToken</see> is canceled.</summary>
+        <returns>The <see cref="T:System.Threading.CancellationTokenRegistration" /> instance that can be used to unregister the callback.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+If this token is already in the canceled state, the delegate will be run immediately and synchronously. Any exception the delegate
+            generates will be propagated out of this method call. <xref:System.Threading.ExecutionContext> is not captured nor flowed to the callback's invocation.
```
Comment (gewarren) [inline xml/System.Threading/CancellationToken.xml:1040]:
```suggestion
generates is propagated out of this method call. <xref:System.Threading.ExecutionContext> is not captured or flowed to the callback's invocation.
```

File: xml/System.Threading/EventWaitHandleAcl.xml [hunk 4]
```diff
@@ -100,11 +101,19 @@ There was some other error. The `HResult` property may provide more information.
         <Parameter Name="rights" Type="System.Security.AccessControl.EventWaitHandleRights" Index="1" FrameworkAlternate="dotnet-plat-ext-6.0;net-6.0" />
       </Parameters>
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="rights">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the event wait handle to be opened. If it's prefixed by "Global", it refers to a machine-wide event wait handle. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide event wait handle. Both prefix and name are case-sensitive.</param>
+        <param name="rights">The desired access rights to apply to the returned event wait handle.</param>
+        <summary>Opens a specified named event wait handle, if it already exists, applying the desired access rights.</summary>
```
Comment (gewarren) [inline xml/System.Threading/EventWaitHandleAcl.xml:106]:
```suggestion
        <summary>Opens a specified named event wait handle, if it already exists, and applies the desired access rights.</summary>
```

File: xml/System.Threading/EventWaitHandleAcl.xml [hunk 5]
```diff
@@ -128,12 +137,19 @@ There was some other error. The `HResult` property may provide more information.
...
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="rights">To be added.</param>
-        <param name="result">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the event wait handle to be opened. If it's prefixed by "Global", it refers to a machine-wide event wait handle. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide event wait handle. Both prefix and name are case-sensitive.</param>
+        <param name="rights">The desired access rights to apply to the returned event wait handle.</param>
+        <param name="result">When this method returns <see langword="true" />, contains an object that represents the named event wait handle if the call succeeded, or <see langword="null" /> otherwise. This parameter is treated as uninitialized.</param>
+        <summary>Tries to open a specified named event wait handle, if it already exists, applying the desired access rights, and returns a value that indicates whether the operation succeeded.</summary>
```
Comment (gewarren) [inline xml/System.Threading/EventWaitHandleAcl.xml:143]:
```suggestion
        <summary>Tries to open a specified named event wait handle, if it already exists, applies the desired access rights, and returns a value that indicates whether the operation succeeded.</summary>
```

File: xml/System.Threading/CancellationToken.xml [hunk 6]
```diff
@@ -778,11 +778,23 @@ For the definition of equality, see the <xref:System.Threading.CancellationToken
...
+        <param name="state">The state to pass to the <paramref name="callback" /> when the delegate is invoked.  This may be <see langword="null" />.</param>
+        <summary>Registers a delegate that will be called when this <see cref="T:System.Threading.CancellationToken">CancellationToken</see> is canceled.</summary>
+        <returns>The <see cref="T:System.Threading.CancellationTokenRegistration" /> instance that can be used to unregister the callback.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+If this token is already in the canceled state, the delegate will be run immediately and synchronously. Any exception the delegate
+            generates will be propagated out of this method call. The current <xref:System.Threading.ExecutionContext>, if one exists,
```
Comment (gewarren) [inline xml/System.Threading/CancellationToken.xml:791]:
```suggestion
generates will be propagated out of this method call. The current <xref:System.Threading.ExecutionContext>, if one exists,
```

File: xml/System.Threading/MutexAcl.xml [hunk 7]
```diff
@@ -92,11 +92,19 @@ There was some other error. The `HResult` property may provide more information.
         <Parameter Name="rights" Type="System.Security.AccessControl.MutexRights" Index="1" FrameworkAlternate="dotnet-plat-ext-6.0;net-6.0" />
       </Parameters>
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="rights">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the mutex to be opened. If it's prefixed by "Global", it refers to a machine-wide mutex. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide mutex. Both prefix and name are case-sensitive.</param>
+        <param name="rights">The desired access rights to apply to the returned mutex.</param>
+        <summary>Opens a specified named mutex, if it already exists, applying the desired access rights.</summary>
```
Comment (gewarren) [inline xml/System.Threading/MutexAcl.xml:97]:
```suggestion
        <summary>Opens a specified named mutex, if it already exists, and applies the desired access rights.</summary>
```

File: xml/System.Threading/MutexAcl.xml [hunk 8]
```diff
@@ -120,12 +128,19 @@ There was some other error. The `HResult` property may provide more information.
...
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="rights">To be added.</param>
-        <param name="result">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the mutex to be opened. If it's prefixed by "Global", it refers to a machine-wide mutex. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide mutex. Both prefix and name are case-sensitive.</param>
+        <param name="rights">The desired access rights to apply to the returned mutex.</param>
+        <param name="result">When this method returns <see langword="true" />, contains an object that represents the named mutex if the call succeeded, or <see langword="null" /> otherwise. This parameter is treated as uninitialized.</param>
+        <summary>Tries to open a specified named mutex, if it already exists, applying the desired access rights, and returns a value that indicates whether the operation succeeded.</summary>
```
Comment (gewarren) [inline xml/System.Threading/MutexAcl.xml:134]:
```suggestion
        <summary>Tries to open a specified named mutex, if it already exists, applies the desired access rights, and returns a value that indicates whether the operation succeeded.</summary>
```

File: xml/System.Threading/SemaphoreAcl.xml [hunk 9]
```diff
@@ -102,11 +102,19 @@ There was some other error. The `HResult` property may provide more information.
         <Parameter Name="rights" Type="System.Security.AccessControl.SemaphoreRights" Index="1" FrameworkAlternate="dotnet-plat-ext-6.0;net-6.0" />
       </Parameters>
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="rights">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the semaphore to be opened. If it's prefixed by "Global", it refers to a machine-wide semaphore. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide semaphore. Both prefix and name are case-sensitive.</param>
+        <param name="rights">The desired access rights to apply to the returned semaphore.</param>
+        <summary>Opens a specified named semaphore, if it already exists, applying the desired access rights.</summary>
```
Comment (gewarren) [inline xml/System.Threading/SemaphoreAcl.xml:107]:
```suggestion
        <summary>Opens a specified named semaphore, if it already exists, and applies the desired access rights.</summary>
```

File: xml/System.Threading/SemaphoreAcl.xml [hunk 10]
```diff
@@ -102,11 +102,19 @@ There was some other error. The `HResult` property may provide more information.
...
+        <returns>An existing named semaphore.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">
+          <paramref name="name" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentException">
+          <paramref name="name" /> is an empty string.</exception>
+        <exception cref="T:System.Threading.WaitHandleCannotBeOpenedException">The named semaphore does not exist or is invalid.</exception>
+        <exception cref="T:System.IO.IOException">The path was not found.
+            -or- A Win32 error occurred.</exception>
```
Comment (gewarren) [inline xml/System.Threading/SemaphoreAcl.xml:116]:
```suggestion

-or-

A Win32 error occurred.</exception>
```

File: xml/System.Threading/SemaphoreAcl.xml [hunk 11]
```diff
@@ -130,12 +138,19 @@ There was some other error. The `HResult` property may provide more information.
...
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="rights">To be added.</param>
-        <param name="result">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="name">The name of the semaphore to be opened. If it's prefixed by "Global", it refers to a machine-wide semaphore. If it's prefixed by "Local", or doesn't have a prefix, it refers to a session-wide semaphore. Both prefix and name are case-sensitive.</param>
+        <param name="rights">The desired access rights to apply to the returned semaphore.</param>
+        <param name="result">When this method returns <see langword="true" />, contains an object that represents the named semaphore if the call succeeded, or <see langword="null" /> otherwise. This parameter is treated as uninitialized.</param>
+        <summary>Tries to open a specified named semaphore, if it already exists, applying the desired access rights, and returns a value that indicates whether the operation succeeded.</summary>
```
Comment (gewarren) [inline xml/System.Threading/SemaphoreAcl.xml:144]:
```suggestion
        <summary>Tries to open a specified named semaphore, if it already exists, applies the desired access rights, and returns a value that indicates whether the operation succeeded.</summary>
```

File: xml/System.Threading/Thread.xml [hunk 12]
```diff
@@ -4108,8 +4108,19 @@ When you call the `Suspend` method on a thread, the system notes that a thread s
...
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <summary>Causes the operating system to change the state of the current instance to <see cref="F:System.Threading.ThreadState.Running" />.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Unlike <xref:System.Threading.Thread.Start(System.Object)>, which captures the current <xref:System.Threading.Thread.ExecutionContext> and uses that context to invoke the thread's delegate,
+            <xref:System.Threading.Thread.UnsafeStart(System.Object)> explicitly avoids capturing the current context and flowing it to the invocation.
```
Comment (gewarren) [inline xml/System.Threading/Thread.xml:4118]:
```suggestion
<xref:System.Threading.Thread.UnsafeStart(System.Object)> explicitly avoids capturing the current context and flowing it to the invocation.
```

File: xml/System.Threading/Thread.xml [hunk 13]
```diff
@@ -4143,9 +4154,21 @@ When you call the `Suspend` method on a thread, the system notes that a thread s
...
-        <remarks>To be added.</remarks>
+        <param name="parameter">An object that contains data to be used by the method the thread executes.</param>
+        <summary>Causes the operating system to change the state of the current instance to <see cref="F:System.Threading.ThreadState.Running" />, and optionally supplies an object containing data to be used by the method the thread executes.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Unlike <xref:System.Threading.Thread.Start(System.Object)>, which captures the current <xref:System.Threading.Thread.ExecutionContext> and uses that context to invoke the thread's delegate,
+            <xref:System.Threading.Thread.UnsafeStart(System.Object)> explicitly avoids capturing the current context and flowing it to the invocation.
```
Comment (gewarren) [inline xml/System.Threading/Thread.xml:4165]:
```suggestion
<xref:System.Threading.Thread.UnsafeStart(System.Object)> explicitly avoids capturing the current context and flowing it to the invocation.
```

**Review by gewarren — APPROVED** [review]
Left some minor suggestions. Thanks for adding these!

**Comment by kouvel** [comment]
Thanks!

**Pull request merged**